packages/desktop-client/src/components/accounts/Account.js

```diff
@@ -523,11 +551,27 @@ class AccountInternal extends PureComponent {
           this.props.savePrefs({ ['show-balances-' + accountId]: false });
           this.setState({ showBalances: false, balances: [] });
         } else {
+          this.setState({
+            transactions: [],
+            transactionCount: 0,
+            filters: [],
+            search: '',
+            sort: [],
+            showBalances: true,
+          });
+          this.fetchTransactions();
           this.props.savePrefs({ ['show-balances-' + accountId]: true });
-          this.setState({ showBalances: true });
           this.calculateBalances();
         }
         break;
+      case 'remove-sorting':
+        let filters = this.state.filters;
+        this.setState({ sort: [] });
+        filters.length > 0
+          ? this.applyFilters([...filters])
+          : this.fetchTransactions();
+        this.state.search !== '' && this.onSearch(this.state.search);
+        break;
```

Comment:
Since this isn’t an expression can you expand it to an `if` statement? I don’t think we really use the ternary pattern like this in the rest of the codebase. Also, it’s best to use an explicit block to avoid issues if a different `case` wants to declare a `filters` variable:
```suggestion
      case 'remove-sorting': {
        let filters = this.state.filters;
        this.setState({ sort: [] });
        if (filters.length > 0) {
          this.applyFilters([...filters])
        } else {
          this.fetchTransactions();
        }
        this.state.search !== '' && this.onSearch(this.state.search);
        break;
      }
```